### DISCUSSION TITLE
Enhanced toggle by adding a tag property in the segments structure

### DISCUSSION THREAD

**Discussion by cmaahs** [JanDeDobbeleer/oh-my-posh#3503] [Ideas]
I recently re-discovered oh-my-posh!  I've even already written my own segment to show me the number of MR/PR when I am at the top directory of a git worktree... Of course, it is horribly lacking caching currently, and the call is visibly slow.  So, toggle... 

I use toggle quite a bit, and in this case, easy enough, toggle based on the `type`.  I also added several text based segments, and have worked around the full toggle of text type with env variable checking.  It occurs to me that if a `tag` property was added to the `segment` object, then a toggle could optionally operate on the `tag` value rather than the `type`.  

Of course, easy in my head... :)

**Comment by JanDeDobbeleer** [comment]
@cmaahs perfectly solvable I guess. I'll have a look after lunch.

EDIT: we already have `Alias` which can be used for this purpose, I will extend the logic so it validates on Alias too.

👉🏻  https://github.com/JanDeDobbeleer/oh-my-posh/pull/3504

**Reply by cmaahs** [reply]
Excellent.  I clearly hadn't discovered the `Alias` feature yet...
